File: 13. Roman to Integer Solved.cpp
// 13. Roman to Integer Solved // c++

class Solution {
public:
    int romanToInt(string s) {
        unordered_map<char, int> s_dict = {
            {'I', 1},
            {'V', 5},
            {'X', 10},
            {'L', 50},
            {'C', 100},
            {'D', 500},
            {'M', 1000},
        };
        int sum = s_dict[s[s.size()-1]];
        for (int i = 0; i < s.size()-1; i++){
            if (s_dict[s[i]]< s_dict[s[i+1]]){
                sum -= s_dict[s[i]];
            }
            else {
                sum += s_dict[s[i]];
            }
        }
        return sum;
    }
};

File: 13. Roman to Integer Solved.py
# 13. Roman to Integer Solved # python

class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        s_dict = {
                'I': 1,
                'V': 5,
                'X': 10,
                'L': 50,
                'C': 100,
                'D': 500,
                'M': 1000,
        }
        sum = s_dict[s[len(s)-1]]
        for i in range(len(s)-1):
            if s_dict[s[i]]< s_dict[s[i+1]]:
                sum -= s_dict[s[i]]
            else:
                sum += s_dict[s[i]]
        return sum